### ISSUE TITLE
About the output of the action during evaluation episode

### ISSUE THREAD

**Issue by AdrienLin1** [nicklashansen/tdmpc2#79]
Hi, thanks for your code. I have some questions about evalution episode, shouldn't only use the output of pi of the agent? Why you still use the ''plan'' function to output the actions? Maybe it's relative fair to use only the policy network's ouput.
``` python
	def act(self, obs, t0=False, eval_mode=False, task=None):
		"""
		Select an action by planning in the latent space of the world model.

		Args:
			obs (torch.Tensor): Observation from the environment.
			t0 (bool): Whether this is the first observation in the episode.
			eval_mode (bool): Whether to use the mean of the action distribution.
			task (int): Task index (only used for multi-task experiments).

		Returns:
			torch.Tensor: Action to take in the environment.
		"""
		obs = obs.to(self.device, non_blocking=True).unsqueeze(0)
		if task is not None:
			task = torch.tensor([task], device=self.device)
        # always using the function during evalution episode
		if self.cfg.mpc:
			return self.plan(obs, t0=t0, eval_mode=eval_mode, task=task).cpu()
		z = self.model.encode(obs, task)
		action, info = self.model.pi(z, task)
		if eval_mode:
			action = info["mean"]
		return action[0].cpu()
```

**Comment by nicklashansen** [comment]
Hi @AdrienLin1, thanks for your interest in our work! I am not sure I follow, why would it not be ok to use planning? Empirically planning appears to perform better than a simple parametric policy, especially in high-dim tasks or multi-task settings.
